src/data/array_interface.h

```diff
@@ -400,6 +401,13 @@ class ArrayInterface {
 
     data = ArrayInterfaceHandler::ExtractData(array, n);
     static_assert(allow_mask ? D == 1 : D >= 1, "Masked ndarray is not supported.");
+
+    this->DispatchCall([&](auto const *values) {
```

Comment:
```suggestion
    this->DispatchCall([&](auto const *data_casted) {
```
to indicate that the parameter is really a casted version of `data`.

Reply:
@hcho3 Not casted version is `void const*`, which is actually casted from input pointer (user-input-ptr -> void const* -> user-input-ptr)

Reply:
How about `data_typed_ptr`?

Reply:
sounds good.

Reply:
done.